cnf-certification-test/preflight/suite.go

```diff
@@ -123,38 +147,42 @@ func generatePreflightContainerGinkgoTest(testName, description, suggestion stri
 		identifiers.NonTelco: identifiers.Optional,
 		identifiers.Extended: identifiers.Optional,
 	}, identifiers.TagPreflight)
-	testID, tags := identifiers.GetGinkgoTestIDAndLabels(aID)
 
-	// Start the ginkgo It block
-	ginkgo.It(testID, ginkgo.Label(tags...), func() {
-		// Collect all of the failed and errored containers
-		var failedContainers []string
-		var erroredContainers []string
-		for _, cut := range containers {
-			for _, r := range cut.PreflightResults.Passed {
-				if r.Name() == testName {
-					logrus.Infof("%s has passed preflight test: %s", cut.String(), testName)
+	testID, tags := identifiers.GetGinkgoTestIDAndLabels(aID)
+	check := checksdb.NewCheck(testID, tags).
+		WithSkipCheckFn(testhelper.GetNoContainersUnderTestSkipFn(&env)).
```

Comment:
Why not add the docker check here as a check function? This way the skip reason will appear in the claim file for each tests scoped by the label

Reply:
That'd be great, but It's not that simple with the current code. That was my first approach, but when I tested it, running the preflight ts without the PFTL_DOCKERCONFIG env var set, preflight would fail to download containers of some of the discovered operators (e.g. openshift's ones), [returning error](https://github.com/test-network-function/cnf-certification-test/blob/ae2981eb3e58c0012abe32636796f4de5d4433d8/pkg/provider/containers.go#L127) on that preflight's container check. But that is happening [before](https://github.com/test-network-function/cnf-certification-test/blob/ae2981eb3e58c0012abe32636796f4de5d4433d8/cnf-certification-test/preflight/suite.go#L102) we can create our own check for it.

I'm not even sure that it can be refactored to add that container to the nonCompliant list when that happens, because when preflight's [check.Run()](https://github.com/test-network-function/cnf-certification-test/blob/ae2981eb3e58c0012abe32636796f4de5d4433d8/pkg/provider/containers.go#L123) function returns error, you shouldn't expect the preflight's Result struct to be properly [initialized](https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/31f7e815872037c910b57e16a50489cb1e566174/container/check_container.go#L42). But we need that struct's fields to create our internal test identifier that is used afterwards to generate our checksdb.Check.

Anyways, I prefer to limit the scope of this PR to the minimum set of required changes to migrate this test suite as it is right now. We can try addressing that refactor in following PRs.